File: Worker.js
const { runInContext } = require("vm");
const zmq=require("zeromq");
const sock = new zmq.Pull();

runInContext();
async function run(){
    await sock.connect("tcp://127.0.0.1:7000");
    console.log("Connected to Server");

    for await ( const msg of sock){
        console.log('received job ${msg.toString()}');
        
    }
}

File: worker.py
import zmq
import keyboard

c = zmq.Context()
s = c.socket(zmq.PULL) # Socket type PULL
s.connect("tcp://127.0.0.1:7000") # Binding it to port 7000 and tcp address 127.0.0.1 
print("Connection Established ")
while True:
    job = s.recv_json()  # Recieve Taska from server
    print ("Received job no. : ",job)
    
